File: MWR/Source/MWRSolution/MWRDBModule/DBModule/VewTxnHeaderWithCarDispatchCtrl.cs
using System;
using System.Collections.Generic;
using System.Text;
using ComLib.db;

namespace YRKJ.MWR
{
    public class VewTxnHeaderWithCarDispatchCtrl : BaseDataCtrl
    {
        public static bool QueryPage(DataCtrlInfo dcf, SqlWhere sw, int page, int pageSize, ref List<VewTxnHeaderWithCarDispatch> itemList, ref string errMsg)
        {
            SqlQueryMng sqm = new SqlQueryMng();
            sqm.Condition.Where.AddWhere(sw);
            return QueryPage(dcf, sqm, page, pageSize, ref itemList, ref errMsg);
        }
 
        public static bool QueryPage(DataCtrlInfo dcf, SqlQueryMng sqm, int page, int pageSize,ref List<VewTxnHeaderWithCarDispatch> itemList,ref string errMsg)
        {
            try
            {
                string sql = sqm.getVewPageSql(VewTxnHeaderWithCarDispatch.getSql(),page, pageSize);
                SqlCommonFn.DebugLog(sql);
                itemList = SqlDBMng.getInstance().query(sql, new VewTxnHeaderWithCarDispatch(), sqm.getParamsArray());
                if (itemList.Count != 0)
                {
                    dcf.RowCount = itemList[0].TEM_COLUMN_COUNT;
                    dcf.PageCount = ComLib.ComFn.getPageCount(pageSize, dcf.RowCount);
                }
            }
            catch (Exception e)
            {
                errMsg = e.Message;
                return false;
            }
            return true ;
        }

         public static bool QueryMore(DataCtrlInfo dcf, SqlWhere sw,
             ref List<VewTxnHeaderWithCarDispatch> itemList,ref string errMsg)
         {
             SqlQueryMng sqm = new SqlQueryMng();
             sqm.Condition.Where.AddWhere(sw);
             return QueryMore(dcf, sqm,ref itemList,ref errMsg);
         }

         public static bool QueryMore(DataCtrlInfo dcf,SqlQueryMng sqm,
             ref List<VewTxnHeaderWithCarDispatch> itemList,ref string errMsg){
                 if (itemList == null)
                 {
                     itemList = new List<VewTxnHeaderWithCarDispatch>();
                 }
                 itemList = new List<VewTxnHeaderWithCarDispatch>();
                 string sql = sqm.getVewSql(VewTxnHeaderWithCarDispatch.getSql());
                //System.out.println(sql);
                 try {
                     itemList = SqlDBMng.getInstance().query(sql, new VewTxnHeaderWithCarDispatch(),sqm.getParamsArray());
             
                 } catch (Exception e) {
                
                     errMsg = e.Message;
                     return false;
                 }

                 return true;
         }

         public static bool QueryOne(DataCtrlInfo dcf,SqlQueryMng sqm,ref VewTxnHeaderWithCarDispatch item,ref string errMsg){
              List<VewTxnHeaderWithCarDispatch> itemList = new List<VewTxnHeaderWithCarDispatch>();
             if (!QueryMore(dcf, sqm, ref itemList, ref errMsg))
             {
                 return false;
             }

             if (itemList.Count > 0)
             {
                 item = itemList[0];
             }
             else
             {
                 item = null;
             }
             return true;
         }

         public static bool QueryOne(DataCtrlInfo dcf,SqlWhere sw,ref VewTxnHeaderWithCarDispatch item,ref string errMsg){
             List<VewTxnHeaderWithCarDispatch> itemList = new List<VewTxnHeaderWithCarDispatch>();
             if (!QueryMore(dcf, sw, ref itemList, ref errMsg))
             {
                 return false;
             }

             if (itemList.Count > 0)
             {
                 item = itemList[0];
             }
             else
             {
                 item = null;
             }
             return true;
         }
     }
}

File: DemoApp/TblModel/VewTxnHeaderWithCarDispatchCtrl.java
package com.yrkj.ar.bean;
import java.util.List;

import com.yrkj.lib.db.BaseDataCtrl;
import com.yrkj.lib.db.DataCtrlInfo;
import com.yrkj.lib.db.SqlCommonFn;
import com.yrkj.lib.db.SqlDBMng;
import com.yrkj.lib.db.SqlQueryMng;
import com.yrkj.lib.db.SqlUpdateColumn;
import com.yrkj.lib.db.SqlUpdateMng;
import com.yrkj.lib.db.SqlWhere;

public class VewTxnHeaderWithCarDispatchCtrl extends BaseDataCtrl {
    public static List<VewTxnHeaderWithCarDispatch> QueryMore(DataCtrlInfo dcf,SqlWhere sw){
        SqlQueryMng sqm = new SqlQueryMng();
        sqm.Condition.Where.AddWhere(sw);
        return QueryMore(dcf,sqm);
    }

    public static List<VewTxnHeaderWithCarDispatch> QueryMore(DataCtrlInfo dcf,SqlQueryMng sqm){
        
        String sql = sqm.getVewSql(VewTxnHeaderWithCarDispatch.getSql());
        System.out.println(sql);
        try {
            dcf.set(true, "");
            return SqlDBMng.getInstance().query(sql, VewCustomerWithAccount.class,sqm.getParamsArray());
        } catch (Exception e) {
            dcf.set(false, e.getMessage());
            return null;
        }
    }

    public static VewTxnHeaderWithCarDispatch QueryOne(DataCtrlInfo dcf,SqlQueryMng sqm){
        List<VewTxnHeaderWithCarDispatch> itemList = QueryMore(dcf,sqm);
        if(itemList.size() > 0){
            return itemList.get(0);
        }
        return null;
    }

    public static VewTxnHeaderWithCarDispatch QueryOne(DataCtrlInfo dcf,SqlWhere sw){
        List<VewTxnHeaderWithCarDispatch> itemList = QueryMore(dcf,sw);
        if(itemList.size() > 0){
            return itemList.get(0);
        }
        return null;
    }

}
